src/firmware_new/tools/send_proto_over_udp.cpp

```diff
@@ -6,18 +6,36 @@
 #include "boost/bind.hpp"
 #include "firmware_new/boards/frankie_v1/constants.h"
 #include "firmware_new/proto/control.pb.h"
-#include "firmware_new/tools/communication/robot_communicator.h"
-#include "firmware_new/tools/communication/transfer_media/network_medium.h"
 #include "google/protobuf/message.h"
 #include "software/logger/logger.h"
-#include "software/multithreading/thread_safe_buffer.h"
+#include "software/networking/proto_multicast_sender.h"
 
 
 using boost::asio::ip::udp;
 using google::protobuf::Message;
 
+
+/*
+ * send_proto_over_udp
+ *
+ * This file serves as a testing file to easily send proto at a fixed rate.
+ * Plug the computers ethernet port into an ethernet switch/router.
+ *
+ * Run `ifconfig` in the terminal and find your ethernet interface.
+ * On linux it will usually be eth0 or enp3s0f1
+ *
+ * Plug the STM32H7 into the same switch/router and then run this with
+ * bazel run //firmware_new/tools:send_proto_over_udp -- your_interface_here
+ *
+ */
 int main(int argc, char* argv[])
 {
+    if (argc != 2)
+    {
+        throw std::invalid_argument(
+            "Please provide the interface you wish to multicast over");
+    }
```

Comment:
Nice to Have: implement this with <boost/program_options.hpp>

Reply:
looked into this, not sure if its worth having that for this script. this will probably be gone after all the networking stuff is done, we will probably use robot_diagnostics to test the network